content/blog/2019-08-13-popler.md

```diff
@@ -0,0 +1,81 @@
+    ## #   treatment_type_2 <chr>, treatment_type_3 <chr>, lat_lter <dbl>, lng_lter <dbl>, taxas <list>
+
+This line of code shows that there are only four datasets with these characteristics. If interested we can download these directly:
+
+```r
+df_obj <- pplr_get_data( datatype == 'cover' & studytype == 'exp' & duration_years > 20 )
+```
+
+
+
+#### Managing the popler database and its API
+
+One of the biggest hurdles in creating the popler R package was integrating it with an Application Programming Interface (API) that allows secure access to the popler online database. In principle, the popler R package could access the popler online database directly from R using functionalities of R packages such as RPostgreSQL. However, this poses substantial security threats, and in fact our online database did experience a minor security breach. We therefore welcomed the suggestion to use an API by [Noam Ross](authors/noam-ross/), from rOpenSci. Subsequently, [Scott Chamberlain](authors/scott-chamberlain/), also from rOpenSci, was kind enough to develop the API using the Ruby language, the code for which is currently stored on github.
+
+Querying an SQL database using an API improves security, but poses a few limitations on the features of the popler R package, and imposes the additional challenge of managing the API. The API limits the ways in which the R package can query the popler online database. In particular, our API allows querying popler in two ways: downloading the metadata of all 305 datasets, and downloading entire datasets. Fortunately, these two queries reflect the vast majority of operations that users perform when using the popler R package. In particular, the metadata of all 305 datasets is essentially what is used by the `pplr_browse()` function, and users almost always tend to use `pplr_get_data()` to download entire datasets. Hence, the limitation posed by the API were of little importance when weighed against the benefit provided by increased security. As a side note, using an API provides a minor, yet somewhat enticing perk: a progress bar for the downloads of `pplr_get_data()`.
+
+The API also need be managed in order to avoid crashes which make downloads impossible. The API crashes whenever the online database is stopped and restarted, and whenever the API web service crashes. Stopping and re-starting the database occur, for example, when we want changes to the popler online database configuration to take effect. In those rare cases, the API will crash after stopping the database, and it will need to be restared, taking a few hours before being fully functional again. Moreover, the API will stop working whenever its web service crashes. These crashes occasionally happen, and they require restarting the web server of the API. To monitor whether the web service of our API is running, we use [Uptime Robot](https://uptimerobot.com/), a free application that quantifies the time during which the web server is available to users.
```

Comment:
Maybe condense this section a bit: “The API also need be managed in order to avoid crashes…”. Providing details here detracts a bit from the strong message of the rest of the post